Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -84,64 +105,72 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
                 self.tableView.reloadData()
             }
         }
-        
-        if currentRow > 1 {
-            scrollTableViewToIndex(currentRow)
-        }
     }
     
     func scrollTableViewToIndex(index: Int) {
+        guard index > 1 else {
+            return
+        }
         let moveToIndexPath = NSIndexPath(forRow: index-2, inSection: 0)
         self.tableView.reloadRowsAtIndexPaths([moveToIndexPath], withRowAnimation: .None)
         self.tableView.scrollToRowAtIndexPath(moveToIndexPath, atScrollPosition: UITableViewScrollPosition.Middle, animated: false)
     }
     
-    override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
-        self.resizeHeight()
+    override func willTransitionToTraitCollection(newCollection: UITraitCollection, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.willTransitionToTraitCollection(newCollection, withTransitionCoordinator: coordinator)
+        tableView.snp_updateConstraints { make in
+            make.bottom.equalTo(self.view).offset(bvc.shouldShowFooterForTraitCollection(newCollection) ? -UIConstants.ToolbarHeight : 0)
+        }
+    }
+    
+    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+        tableView.snp_updateConstraints { make in
+            make.height.equalTo(min(CGFloat(BackForwardViewUX.RowHeight*listData.count), size.height/2))
+        }
+        if let superView = view.superview {
```

Comment:
If we're setting this view's bounds to the edges either way, do we need to do this check every time we transition size? If we setup the constraints at the start we don't need to reset them.